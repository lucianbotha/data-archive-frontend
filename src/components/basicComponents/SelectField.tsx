import * as React from "react";

interface ISelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {}

const SelectField = (props: ISelectProps) => {
  const { className, children, ...rest } = props;
  return (
    <div className={`control select ${className}`}>
      <select {...rest}>{children}</select>
    </div>
  );
};

export default SelectField;

/**
 * The "any" option. The key attribute has the value "any" and the value
 * attribute has the value "".
 */
export const AnyOption = () => (
  <option key="any" value="">
    Any
  </option>
);
