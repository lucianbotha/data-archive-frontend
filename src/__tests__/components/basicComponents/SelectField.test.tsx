import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import SelectField, {
  AnyOption
} from "../../../components/basicComponents/SelectField";

describe("SelectField", () => {
  const wrapper = mount(
    <SelectField name="name">
      <AnyOption />
      {["aaa", "bbb", "ccc"].map(t => (
        <option key={t} value={t}>
          {t}
        </option>
      ))}
    </SelectField>
  );
  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("always renders a div", () => {
    const divs = wrapper.find("div");
    expect(divs.length).toBeGreaterThan(0);
  });
  it("always contains a select tag.", () => {
    const select = wrapper.find("div.select");
    expect(select.length).toEqual(1);
    const options = wrapper.find("div.select");
    expect(options.children().length).toBeGreaterThan(0);
  });
  it("select tag children are options and greater than 0.", () => {
    const options = wrapper.find("select");
    expect(options.children().get(1).type).toEqual("option");
    expect(options.children().length).toBeGreaterThan(0);
  });
  it("render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <SelectField name="name">
            <AnyOption />
            {["aaa", "bbb", "ccc"].map(t => (
              <option key={t} value={t}>
                {t}
              </option>
            ))}
          </SelectField>
        )
      )
    ).toMatchSnapshot();
  });
});
