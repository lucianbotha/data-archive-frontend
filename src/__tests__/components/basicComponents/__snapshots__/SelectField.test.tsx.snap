// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField render correctly 1`] = `
<div
  className="control  select undefined"
>
  <select
    id="undefined"
  >
    <option
      key="aaa"
    >
      aaa
    </option>
    <option
      key="bbb"
    >
      bbb
    </option>
    <option
      key="ccc"
    >
      ccc
    </option>
  </select>
</div>
`;

exports[`SelectField render correctly 2`] = `
<div
  className="control  select undefined"
>
  <select
    id="name"
    name="name"
    value="value"
  >
    <option
      key="aaa"
    >
      aaa
    </option>
    <option
      key="bbb"
    >
      bbb
    </option>
    <option
      key="ccc"
    >
      ccc
    </option>
  </select>
</div>
`;
